doc/developers/maintainer.rst

```diff
@@ -17,25 +27,30 @@ Before a release
    change log is reasonably well curated. Some tools for these tasks include:
 
    - ``maint_tools/sort_whats_new.py`` can put what's new entries into
-     sections.
+     sections. It's not perfect, and requires manual checking of the changes.
+     If the whats new list if well curated, it may not be necessary.
 
    - The ``maint_tools/whats_missing.sh`` script may be used to identify pull
      requests that were merged but likely missing from What's New.
 
-Preparing a bug-fix-release
-...........................
+Preparing a release PR
+......................
 
-Since any commits to a released branch (e.g. 0.999.X) will automatically update
-the web site documentation, it is best to develop a bug-fix release with a pull
+Since any commits to an existing branch (e.g. 0.999.X) will automatically
+update the web site documentation, it is best to develop a release with a pull
 request in which 0.999.X is the base. It also allows you to keep track of any
 tasks towards release with a TO DO list.
 
-Most development of the bug fix release, and its documentation, should
-happen in master to avoid asynchrony. To select commits from master for use in
-the bug fix (version 0.999.3), you can use::
+Most development of the release, and its documentation, should happen in master
+to avoid asynchrony. To select commits from master for use in the bug fix
+(version 0.999.3), you can use::
 
+    $ # # assuming master and upstream/master are the same
     $ git checkout -b release-0.999.3 master
-    $ git rebase -i 0.999.X
+    $ git rebase -i upstream/0.999.X
+
+It's nice to have a copy of the ``git rebase -i`` log in the PR to help others
+understand what's included.
```

Comment:
Also explain what should and what should not be included during the rebase? My understanding is that:

- to create e.g. 0.21.rc0, we include everything from 0.20.X 
- to create e.g. 0.21.rc1, we only select specific commits from 0.21.rc0. I.e. we include additional bug fixes and minor docs but typically ignore new features that weren't already in rc0.